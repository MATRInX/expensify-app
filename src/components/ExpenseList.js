import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = ({expenses}) => (
    <div>
        {
            expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id}{...expense}/>})
            )
        }    
    </div>
);

const mapStateToProps = ({expenses, filters})=> {
    return{
        expenses: selectExpenses(expenses, filters)     
    }
}; 

export default connect(mapStateToProps)(ExpenseList);

// const ConnectedExpenseList = connect((state)=>{
//     return {
//         expenses: state.expenses
//     };
// })(ExpenseList);

// export default ConnectedExpenseList; 