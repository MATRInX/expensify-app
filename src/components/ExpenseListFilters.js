import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onSelectChange = (e) => {
        if (e.target.value ==='date') {
            this.props.sortByDate()
        } else if (e.target.value ==='amount') {
            this.props.sortByAmount()
        }  
    };
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
         this.setState(() => ({ calendarFocused }));
    };
    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange}/>
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSelectChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    startDateId="startDateId"
                    endDate={this.props.filters.endDate}
                    endDateId="endDateId"
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDisptachToProps = (dispatch, ownProps) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDisptachToProps)(ExpenseListFilters);