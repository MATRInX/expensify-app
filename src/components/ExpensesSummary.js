import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expenses, expensesTotal}) => {
    return (
        <div>
            <p>
                Viewing {expenses.length} {expenses.length === 1 ? 'expense ' : 'expenses '} 
                totaling {numeral(expensesTotal / 100).format('0,0[.]00 $')}
            </p>
        </div>
    );
};

const mapStateToProps = ({expenses, filters}) => ({
    expenses: selectExpenses(expenses, filters),
    expensesTotal: selectExpensesTotal(expenses)
});

export default connect(mapStateToProps)(ExpensesSummary);