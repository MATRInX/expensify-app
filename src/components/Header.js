import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export class Header extends React.Component {
  onClick = () => {
    this.props.startLogout();
  };
  render() {
    return (
      <header>
      <h1>Expensify</h1>
      <NavLink to="/dashboard" activeClassName="is-active">Dashboard </NavLink>
      <NavLink to="/create" activeClassName="is-active">Create new expense </NavLink>      
      <button onClick={this.onClick}>Logout</button>
    </header>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);