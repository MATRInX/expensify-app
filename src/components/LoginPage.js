import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends React.Component {
    onLogin = () => {
        this.props.startLogin();
    };
    render() {
        return (<div>
            <button onClick={this.onLogin}>Login</button>
        </div>)
    };
};

const mapDispatchToProps =  (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);