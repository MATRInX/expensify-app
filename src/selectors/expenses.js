import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        // Dzieki porównaniu z typem, jeśli będzie inny typ warunek będzie
        // prawdziwy i filtracja nie będzie działac bo zwrócone będzie TRUE
        // const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate;
        // const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate;
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = typeof text !== 'string' || expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
        else {
            return 0;
        }
    });
};