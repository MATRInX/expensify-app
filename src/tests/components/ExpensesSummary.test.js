import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} expensesTotal={195}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with 2 expeneses', () => {
    const wrapper = shallow(
    <ExpensesSummary 
        expenses={[expenses[0], expenses[2]]}
        expensesTotal={4695}
    />);
    expect(wrapper).toMatchSnapshot();
});