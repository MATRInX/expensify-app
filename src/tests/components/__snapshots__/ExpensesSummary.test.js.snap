// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense 
    totaling 
    1.95 $
  </p>
</div>
`;

exports[`should correctly render ExpensesSummary with 2 expeneses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses 
    totaling 
    46.95 $
  </p>
</div>
`;
